paasta_tools/frameworks/native_scheduler.py

```diff
@@ -137,24 +158,37 @@ def task_fits(self, offer):
 
         return True
 
-    def need_more_tasks(self, name):
+    def need_more_tasks(self, name, existingTasks={}, scheduledTasks=[]):
         """Returns whether we need to start more tasks."""
         num_have = 0
-        for task, parameters in self.tasks_with_flags.items():
+        for task, parameters in existingTasks.items():
             if self.is_task_new(name, task) and (parameters.mesos_task_state in LIVE_TASK_STATES):
                 num_have += 1
+
+        for task in scheduledTasks:
+            if task.name == name:
+                num_have += 1
+
         return num_have < self.service_config.get_desired_instances()
 
     def get_new_tasks(self, name, tasks):
-        return set([tid for tid in tasks if self.is_task_new(name, tid)])
+        return set(filter(
+            lambda tid:
+                self.is_task_new(name, tid) and
+                self.tasks_with_flags[tid].mesos_task_state in LIVE_TASK_STATES,
+            tasks))
 
     def get_old_tasks(self, name, tasks):
-        return set([tid for tid in tasks if not self.is_task_new(name, tid)])
+        return set(filter(
+            lambda tid:
+                not(self.is_task_new(name, tid)) and
+                self.tasks_with_flags[tid].mesos_task_state in LIVE_TASK_STATES,
+            tasks))
 
     def is_task_new(self, name, tid):
         return tid.startswith("%s." % name)
 
-    def start_task(self, driver, offer):
+    def tasks_for_offer(self, driver, offer):
         """Starts a task using the offer, and subtracts any resources used from the offer."""
```

Comment:
can you update this comment? as I understand it, it doesn't start any tasks - just returns a list of MesosTasks to be run using the offer.